freeipa/src/main/java/com/sequenceiq/freeipa/controller/UserV1Controller.java

```diff
@@ -59,8 +61,9 @@ public SyncOperationStatus synchronizeUser(SynchronizeUserRequest request) {
             default:
                 throw new BadRequestException(String.format("UserCrn %s is not of resoure type USER or MACHINE_USER", userCrn));
         }
+        // TODO: Get the Request Id from header
```

Comment:
requestId is already added to MDC by the `MDCContextFilter`.
In `UmsUsersStateProvider` we've pulled this out to a `Threadlocal`. 

The benefit of taking it out of the MDC is that the MDC is already propagated to all the relevant threads (i.e., flow, UserService async stuff). The drawback is that we are using the MDC for a non-logging purpose and it's behavior may change.

The right way to do this is to follow the pattern used by redbeams and add a `RequestIdFilter` and a `ThreadLocalRequestIdProvider` classes to the FMS or pull those classes out to the common module. We'll also need to look into the flow code to ensure that the request id is propagated to any threads that flow uses.

Reply:
Taking requestId from MDC, but I don't like unwanted global objects and specially ThreadLocal just for requestId. 
I have moved request Id in the controller and passing from there.